src/extra/components/pulses.py

```diff
@@ -520,7 +520,8 @@ def get_pulse_counts(self, *args, **kwargs):
              DeprecationWarning, stacklevel=2)
         return self.pulse_counts(*args, **kwargs)
 
-    def build_pulse_index(self, pulse_dim='pulseId', include_extra_dims=True):
+    def build_pulse_index(self, pulse_dim='pulseId', include_extra_dims=True,
+                          pulse_dtype=None):
```

Comment:
This feature is added to label the leading and trailing virtual pulses by `-np.inf` and `np.inf`. I tried a couple of different methods, but ended up liking this the most. It's not ideal to have native integers as floats (in the case of `pulseId` or `pulseIndex`), but the alternative would be arbitrary high numbers for the trailing pulse. I tried using `-2` and `-1` respectively, but that became a mess during the alignment process as data is sorted by ToA.